TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ImplConceptAlgoMagnitude.java

```diff
@@ -0,0 +1,129 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.firstinspires.ftc.teamcode;
+
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+import com.qualcomm.robotcore.util.ElapsedTime;
+
+/**
+ * algorithm for calculation of magnitude
+ * for use with advanced mecanum drives
+ * jlemke 9/26/2017 @ 7:26 AM
+ */
+
+@TeleOp(name="Mecanum Magnitude Algorithm fConcept", group="Concept")
+public class ImplConceptAlgoMagnitude extends OpMode
+{
+    // move that gear up
+    Hardware750 robot = new Hardware750();
+    // setup runtime timer
+    private ElapsedTime runtime = new ElapsedTime();
+
+    @Override
+    public void init() {
+        telemetry.addData("Status", "Uninitialized...");
+        robot.init(hardwareMap);
+        telemetry.addData("Status", "Initialized");
+    }
+
+    /*
+     * Code to run REPEATEDLY after the driver hits INIT, but before they hit PLAY
+     */
+    @Override
+    public void init_loop() {
+    }
+
+    @Override
+    public void start() {
+        runtime.reset();
+    }
+
+    @Override
+    public void loop() {
+        telemetry.addData("LeftTrig", gamepad1.left_trigger);
+        telemetry.addData("RightTrig", gamepad1.right_trigger);
+        telemetry.addData("lstick X", gamepad1.left_stick_x);
+        telemetry.addData("lstick Y", (-1 * gamepad1.left_stick_y));
+        double presquare = Math.pow(gamepad1.left_stick_x, 2) + Math.pow(gamepad1.left_stick_y, 2);
+        telemetry.addData("presquare", presquare);
+        double magnitude = ((-1 * gamepad1.left_stick_y) < 0) ? -1*Math.sqrt(presquare) : Math.sqrt(presquare);
+        if (magnitude > 1) {
+            magnitude = 1;
+        }
+        telemetry.addData("magnitude", magnitude);
+        double angle = Math.atan2((-1*gamepad1.left_stick_y),gamepad1.left_stick_x);
+        if (angle < 0) {
+            angle += Math.PI*2;
+        }
+        double leftTrig = (-1 * gamepad1.left_trigger);
+        double rightTrig = gamepad1.right_trigger;
+        telemetry.addData("ang le", angle);
+        telemetry.addData("number of pi", (angle / Math.PI));
+        double extraPi = angle / Math.PI;
+        double VM1 = ((magnitude) * (Math.sin(extraPi + (Math.PI / 4))) + rightTrig);
+        double VM2 = ((magnitude) * (Math.cos(extraPi + (Math.PI / 4))) + leftTrig);
+        double VM3 = ((magnitude) * (Math.cos(extraPi + (Math.PI / 4))) + rightTrig);
+        double VM4 = (((magnitude) * (Math.sin(extraPi + (Math.PI / 4))) + leftTrig));
+        boolean [] VM = new boolean [4];
+        double [] thing = {VM1, VM2, VM3, VM4};
+        for (int i = 1; i < 3; i++) {
```

Comment:
I assume this section is iterating as the flowchart describes, but the lack of comments makes this a little hard to read.